src/main/java/org/radarcns/phone/PhoneSensorManager.java

```diff
@@ -177,7 +246,56 @@ public void processLight(SensorEvent event) {
         send(lightTable, new PhoneLight(time, timeReceived, lightValue));
     }
 
-    public void processBatteryStatus(Intent intent) {
+    private void processGyroscope(SensorEvent event) {
+        // Not normalized axis of rotation in rad/s
+        float axisX = event.values[0];
+        float axisY = event.values[1];
+        float axisZ = event.values[2];
+
+        double timeReceived = System.currentTimeMillis() / 1_000d;
+
+        // nanoseconds uptime to seconds utc
+        double time = ( timeReceived - (System.nanoTime() - event.timestamp) / 1_000_000_000d );
+        send(gyroscopeTable, new PhoneGyroscope(time, timeReceived, axisX, axisY, axisZ));
+    }
+
+    private void processMagneticField(SensorEvent event) {
+        // Magnetic field in microTesla
+        float axisX = event.values[0];
+        float axisY = event.values[1];
+        float axisZ = event.values[2];
+
+        double timeReceived = System.currentTimeMillis() / 1_000d;
+
+        // nanoseconds uptime to seconds utc
+        double time = ( timeReceived - (System.nanoTime() - event.timestamp) / 1_000_000_000d );
```

Comment:
this operation is done quite a few times. Perhaps use a `private static double` function?